web/src/engine/osk/src/visualKeyboard.ts

```diff
@@ -1626,13 +1626,30 @@ export default class VisualKeyboard extends EventEmitter<EventMap> implements Ke
     this.layerLocked = enable;
   }
 
-  raiseKeyEvent(keyEvent: KeyEvent, e: KeyElement): KeyRuleEffects {
+  raiseKeyEvent(keyEvent: KeyEvent, keySpec: ActiveKeyBase, e: KeyElement): KeyRuleEffects {
     // Exclude menu and OSK hide keys from normal click processing
     if(keyEvent.kName == 'K_LOPT' || keyEvent.kName == 'K_ROPT') {
       this.optionKey(e, keyEvent.kName, true);
       return {};
     }
 
+    // Ensure that no matter what, the first key in the distribution matches the "true" key
+    // that the user actually touched.  Even if it's not the most likely key.
+    const keyDistribution = keyEvent.keyDistribution;
+    if(keyDistribution && keyDistribution.length > 1) {
+      const matchIndex = keyDistribution.findIndex(keySample => keySample.keySpec == keySpec);
+      if(matchIndex < 0) {
+        console.error("Could not find and prioritize output key in its fat-finger distribution");
```

Comment:
Will this impact mobile apps? Could it be a warning instead?

Reply:
It would trigger an error notice from the Android app, but that also means that _we_ get notice that something is going wrong.  I'm not 100% sure that we actually get warnings - sometimes User Tests report warnings but nothing shows on Sentry.